packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -300,11 +297,71 @@ export class IamResource {
   }
 }
 
+/**
+ * Attributes for GraphQL From
+ */
+export interface GraphQLApiAttributes {
+  /**
+   * the arn for the GraphQL Api
+   */
+  readonly arn: string,
+
+  /**
+   * the api id of the GraphQL Api
+   */
+  readonly apiId: string,
+}
+
 /**
  * An AppSync GraphQL API
+ *
+ * @resource AWS::AppSync::GraphQLApi
  */
-export class GraphQLApi extends Construct {
+export class GraphQLApi extends GraphQLApiBase {
+  /**
+   * Import a GraphQL API given an arn
+   * @param scope scope
+   * @param id id
+   * @param graphQLApiId the apiId of the api
+   */
+  public static fromGraphQLApiId(scope: Construct, id: string, graphQLApiId: string): IGraphQLApi {
+    const generatedArn = Stack.of(scope).formatArn({
+      service: 'appsync',
+      resource: `apis/${graphQLApiId}`,
+    });
+    return GraphQLApi.fromGraphQLApiAttributes(scope, id, { arn: generatedArn, apiId: graphQLApiId});
+  }
+
+  /**
+   * Import a GraphQL API given an arn
+   * @param scope scope
+   * @param id id
+   * @param graphQLApiArn the arn of the api
+   */
+  public static fromGraphQLApiArn(scope: Construct, id: string, graphQLApiArn: string): IGraphQLApi {
+    const apiId = Fn.select(1, Fn.split('/', graphQLApiArn));
+    if (apiId == undefined || apiId == graphQLApiArn) {
+      throw Error('Arn does not contain a valid apiId');
+    }
+    return GraphQLApi.fromGraphQLApiAttributes(scope, id, { arn: graphQLApiArn, apiId: apiId});
+  }
 
+  /**
+   * Import a GraphQL API through this function
+   * @param scope scope
+   * @param id id
+   * @param attrs GraphQL API Attributes of an API
+   */
+  public static fromGraphQLApiAttributes(scope: Construct, id: string, attrs: GraphQLApiAttributes): IGraphQLApi {
+    class Import extends GraphQLApiBase {
+      public readonly apiId = attrs.apiId;
+      public readonly arn = attrs.arn;
+      constructor (s: Construct, i: string){
+        super(s, i);
```

Comment:
avoid abbreviating - it's more readable to spell out scope and id

Reply:
i cant do scope and id again because it is already declared in as params for the function itself

Reply:
gotcha, let's leave it for now in that case